internal/tools/aws/database.go

```diff
@@ -228,14 +230,85 @@ func (d *RDSDatabase) rdsDatabaseProvision(installationID string, logger log.Fie
 	} else {
-		keyMetadata, err = d.client.kmsCreateSymmetricKey(KMSKeyDescriptionRDS(awsID), []*kms.Tag{
+		caller, err := d.client.Service().sts.GetCallerIdentity(&sts.GetCallerIdentityInput{})
+		if err != nil {
+			return errors.Wrapf(err, "unable to get caller identity when creating RDS DB cluster %s", awsID)
+		}
+
+		callerARN, err := arn.Parse(*caller.Arn)
+		if err != nil {
+			return errors.Wrapf(err, "unable to parse caller ARN when creating RDS DB %s", awsID)
+		}
+
+		// This assumes that the root is at the same account of the caller.
+		rootARN := arn.ARN{
+			AccountID: callerARN.AccountID,
+			Service:   "iam",
+			Partition: "aws",
+			Resource:  "root",
+		}
+
+		// CMKs have their access restricted to users and actions defined in the key itself. Adding the
+		// root user to the policy prevents keys to become unmanageable in the future. In the policy
+		// level we need to give full permissions to the root principal so it can manage the key at
+		// any time if necessary. To other principals, we only allow actions necessary for performing
+		// a designated task. Note that the caller (or provisioner in this case) requires writting
+		// permissions to the key even if this actions are not defined in the user's IAM policy.
+		policyDocument := model.PolicyDocument{
+			Version: model.PolicyDocumentVersion,
+			Statement: []model.StatementEntry{
+				{
+					Sid:    "Allow full control of the key",
+					Effect: model.PolicyStatementEffectAllow,
+					Principal: map[string]string{
+						"AWS": rootARN.String(),
+					},
+					Action: []string{
+						"kms:*",
+					},
+					Resource: model.PolicyStatementAllResources,
+				},
+				{
+					Sid:    "Allow administration of the key",
+					Effect: model.PolicyStatementEffectAllow,
+					Principal: map[string]string{
+						"AWS": callerARN.String(),
+					},
+					Action: []string{
+						"kms:Create*",
+						"kms:Describe*",
+						"kms:Enable*",
+						"kms:List*",
+						"kms:Put*",
+						"kms:Update*",
+						"kms:Revoke*",
+						"kms:Disable*",
+						"kms:Get*",
+						"kms:Delete*",
+						"kms:ScheduleKeyDeletion",
+						"kms:CancelKeyDeletion",
+					},
+					Resource: model.PolicyStatementAllResources,
+				},
+			},
+		}
+
+		policyDocumentBytes, err := policyDocument.Marshal()
+		if err != nil {
+			return errors.Wrapf(err, "unable to create a CMK policy document for encrypting RDS DB cluster %s", awsID)
+		}
+
+		keyPolicyDocument := string(policyDocumentBytes)
+		keyTags := []*kms.Tag{
 			{
 				TagKey:   aws.String(DefaultRDSEncryptionTagKey),
 				TagValue: aws.String(awsID),
 			},
-		})
+		}
+		keyMetadata, err = d.client.kmsCreateSymmetricKey(KMSKeyDescriptionRDS(awsID), keyTags, &keyPolicyDocument)
 		if err != nil {
-			return errors.Wrapf(err, "unable to create an encryption key for db cluster %s", awsID)
+			return errors.Wrapf(err,
+				"unable to create an encryption key for RDS DB cluster %s: key policy: %s", awsID, keyPolicyDocument)
```

Comment:
Hmm, print the whole key policy document in the error? I would ask if we are sure we would want to do that or if this could be handled in a debug level message?

Reply:
What was exactly your concern? I'm answering this questions because we may need this PR in the future.